File: backend/app/schemas/roadmap.py
from typing import Literal

from pydantic import BaseModel

RoadmapPriority = Literal["high", "medium", "low"]


class RoadmapItemResponse(BaseModel):
    id: str
    title: str
    reason: str
    priority: RoadmapPriority
    missing_skills: list[str]
    related_skills: list[str]


class RoadmapResponse(BaseModel):
    items: list[RoadmapItemResponse]

File: frontend/lib/api/types/roadmap.ts
export type RoadmapPriority = "high" | "medium" | "low";

export type RoadmapItem = {
  id: string;
  title: string;
  reason: string;
  priority: RoadmapPriority;
  missing_skills: string[];
  related_skills: string[];
};

export type RoadmapResponse = {
  items: RoadmapItem[];
};
